### PR TITLE
Implemented groomer login and modified user/groomer registration logic

### PR DESCRIPTION
**Pull request by Shinsaegae**
Added logic for groomers to be able to log in (but no groomer home screen yet).

Modified account creation logic slightly to enable users to create both a user and groomer account with a single email.
Also slightly modified UI to accommodate for this.

These don't yet work with Google authentication.

### CODE REVIEW

**Comment by Shinsaegae** [comment]
Still need to fix some merging issues

**Comment by Chocolat2** [comment]
Kotlin Formatting checks have failed, I am currently looking into that to try and resolve the issue.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt [hunk 1]
```diff
@@ -68,6 +68,9 @@ import com.android.PetPamper.model.Groomer
 import com.android.PetPamper.model.GroomerReviews
 import com.android.PetPamper.model.LocationMap
 import com.android.PetPamper.model.User
+import com.android.PetPamper.ui.screen.CustomTextButton
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.firestore
 
 class GroomerSignUpViewModel {
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt:74]: ViewModel component of the MVVM design pattern is usually in a separate file to enforce separation between the UI, the data formatting and the data processing.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt [hunk 2]
```diff
@@ -87,20 +90,33 @@ const val NUM_STEPS = 12
...
 @Composable
 fun GroomerRegister(viewModel: GroomerSignUpViewModel, navController: NavController) {
+    val firebaseConnection = FirebaseConnection()
+    val db = Firebase.firestore
+
     var currentStep by remember { mutableIntStateOf(1) }
 
+    var registeredAsUser by remember { mutableStateOf(false) }
+
     when (currentStep) {
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt]: A match-like structure to handle screen UI changes seems weird to me, is it a common practice on Kotlin/Andriod projects ?

File: app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt [hunk 3]
```diff
@@ -73,8 +75,12 @@ import androidx.navigation.compose.rememberNavController
...
 import com.android.PetPamper.database.FirebaseConnection
 import com.android.PetPamper.model.Address
+import com.android.PetPamper.model.Groomer
 import com.android.PetPamper.model.LocationMap
 import com.android.PetPamper.model.User
+import com.android.PetPamper.ui.screen.CustomTextButton
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.firestore
 
 class SignUpViewModel {
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt:76]: ViewModel component of the MVVM design pattern is usually in a separate file to enforce separation between the UI, the data formatting and the data processing.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt [hunk 4]
```diff
@@ -88,20 +94,33 @@ class SignUpViewModel {
 
 @Composable
 fun Register(currentStep1: Int, viewModel: SignUpViewModel, navController: NavController) {
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt:87]: Register.kt seems to be a general implementation to be reused and extended for each specific case, yet GroomerRegister.kt implements another similar logic entirely. Could these be refactored ?

**Review by Chocolat2 — CHANGES_REQUESTED** [review]
Overall code seems to be working fine on the emulator, however there are no tests designed for the new feature. Additionally, the improvements/modifications on the registration feature may need new or modified tests.

**Review by chtidaba — APPROVED** [review]
Code now passes the CI correctly, it successfully implement the requested functionalities.

Still some little bugs that does not affect the user experience but when registering a user, in the logcat we have an error suggesting that it tries to register the user multiple times. But bug now is fixed. 

for the rest, nothing to suggest, LGTM!

File: app/src/main/java/com/android/PetPamper/ui/screen/LoginScreen.kt [hunk 5]
```diff
@@ -176,85 +196,137 @@ fun SignIn(navController: NavHostController) {
...
+                    )
+                    Text(
+                        text = if (isGroomer) "I am a groomer" else "I am a user",
+                        style = TextStyle(
+                            textAlign = TextAlign.Center,
+                            fontSize = 16.sp, fontWeight = FontWeight(600)
+                        ),
+                        modifier = Modifier.offset(x=110.dp)
+                    )
+//                      Button(
```
[the remark this answers is not in the document]
Comment (Shinsaegae) [reply app/src/main/java/com/android/PetPamper/ui/screen/LoginScreen.kt]: Removed the commented code

File: app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt (hunk 2 above)
Comment (Chocolat2) [reply app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt]: The UI change could be handled more efficiently by refactoring the navigation throughout the app, see:
https://developer.android.com/develop/ui/compose/navigation
https://developer.android.com/develop/ui/compose/architecture
https://developer.android.com/guide/navigation
https://github.com/android/nowinandroid

File: app/src/main/java/com/android/PetPamper/ui/screen/LoginScreen.kt [hunk 6]
```diff
@@ -176,15 +184,38 @@ fun SignIn(navController: NavHostController) {
 
                 Button(
                     onClick = {
+                      errorMessage = "Login failed, email or password is incorrect"
                       if (email.isBlank() || password.isBlank()) {
                         login = false
                       } else {
                         firebaseConnection.loginUser(
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/LoginScreen.kt:191]: UI events should be handled in the ViewModel, which itself should delegate data queries and updates to the Model and the data layer.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt [hunk 7]
```diff
@@ -86,10 +89,19 @@ const val NUM_STEPS = 12
 
 @Composable
 fun GroomerRegister(viewModel: GroomerSignUpViewModel, navController: NavController) {
+  val firebaseConnection = FirebaseConnection()
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/GroomerRegister.kt:92]: UI events should be handled in the ViewModel, which itself should delegate data queries and updates to the Model and the data layer.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt [hunk 8]
```diff
@@ -79,10 +84,20 @@ class SignUpViewModel {
...
+  val firebaseConnection = FirebaseConnection()
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt:87]: UI events should be handled in the ViewModel, which itself should delegate data queries and updates to the Model and the data layer.

File: app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt [hunk 9]
```diff
@@ -221,6 +255,113 @@ fun Register(currentStep1: Int, viewModel: SignUpViewModel, navController: NavCo
...
+                              .addOnFailureListener { exception ->
+                                login = false
+                                errorMessage = "Login failed, email or password is incorrect"
+                                Log.e("Firebase query", "Get failed with ", exception)
+                              }
+                        },
+                        { login = false })
+                  }
+                },
+                colors = ButtonDefaults.buttonColors(Color(0xFF2491DF)),
```
Comment (Chocolat2) [inline app/src/main/java/com/android/PetPamper/ui/screen/register/Register.kt:359]: Colors values can be specified in R.values.color
Theme for he whole app can be specified in R.values.theme

**Review by Chocolat2 — COMMENTED** [review]
Switch toggle feature looks good, overall app codebase will need to be refactored before adding new features to keep code maintainability and testability.
Test should be implemented more frequently on new (and existing features).

**Review by Chocolat2 — APPROVED** [review]

**Pull request merged**